source/common/filter/config_discovery_impl.h

```diff
@@ -102,16 +102,20 @@ class DynamicFilterConfigProviderImpl : public DynamicFilterConfigProviderImplBa
   }
 
   void onConfigRemoved(Config::ConfigAppliedCb applied_on_all_threads) override {
-    tls_.runOnAllThreads(
-        [config = default_configuration_](OptRef<ThreadLocalConfig> tls) { tls->config_ = config; },
-        [this, applied_on_all_threads]() {
-          // This happens after all workers have discarded the previous config so it can be safely
-          // deleted on the main thread by an update with the new config.
-          this->current_config_ = default_configuration_;
-          if (applied_on_all_threads) {
-            applied_on_all_threads();
-          }
-        });
+    const absl::optional<Envoy::Http::FilterFactoryCb> default_config =
+        default_configuration_ ? absl::make_optional(factory_cb_(default_configuration_.value()))
+                               : absl::nullopt;
+    tls_.runOnAllThreads([config = default_config,
```

Comment:
I'm a bit confused why we capture default_config twice here.

Reply:
Thank you that's a mistake; I think I modified the wrong closure at first